### PR TITLE
Updated to pyinstaller repository

### PR DESCRIPTION
**Pull request by VakarisZ**
Updated our base from 4.0.0dev to 4.2 master. This includes the latest fixes in windows bootloader's memory leaks and re-built bootloaders.

### CODE REVIEW

**Comment by mssalvatore** [comment]
@VakarisZ Are you good with my 3 commits? What's left to do before we merge?

File: bootloader/wscript [hunk 1]
```diff
@@ -761,7 +761,7 @@ def build(ctx):
         # made in the configure phase.
         libs = ['DL', 'M', 'Z',  # 'z' - zlib, 'm' - math,
                 'THR']  # may be used on FreBSD
-        staticlibs = []
+        staticlibs = ['curl']
```
Comment (VakarisZ) [inline bootloader/wscript:764]: Why did you have to add it?

**Review by VakarisZ — COMMENTED** [review]
Looks good, but can you explain why we need curl as a static lib?

**Comment by mssalvatore** [comment]
> Looks good, but can you explain why we need curl as a static lib?

If we don't add it, the curl library doesn't get linked and the build fails.

**Pull request merged**